lib/homework2.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Homework2 do
+  @moduledoc """
+    This module provides a single function `Homework2.send/1`, which spawns a process.
+    This process then receives a message passed as an argument to the function.
+  """
+
+  @doc """
+  Receive a reply from a spawned process:
+
+  ## Examples
+      iex> pid = Homework2.send {:ping, self()}
+      pid
+      iex> assert_receive {:pong, ^pid}
+      {:pong, pid}
+
+      iex> pid = Homework2.send {:boom, self()}
+      pid
+      iex> assert_receive {:error, ^pid}
+      {:error, pid}
+  """
+
+  @spec send({}) :: pid()
+  def send(message) do
+    pid =
+      Process.spawn(
+        fn ->
+          receive do
+            {:ping, pid} -> send(pid, {:pong, self()})
+            {_, pid} -> send(pid, {:error, self()})
+          end
```

Comment:
There is no magic. [`receive/1`](https://hexdocs.pm/elixir/Kernel.SpecialForms.html#receive/1) well, receives the message, and that’s it. If you want to receive another one, you should call yourself recursively. Somewhat alongside
```elixir
receiver =
  fn next_fun -> # the function to return the actual function
    fn -> # this is actual handler 
      receive do
        {:ping, pid} ->
          send(pid, {:pong, self()})
          next_fun.() # call ourselves recursively
        _match_everything_because_nobody_promised_pid_in_malformed_message -> 
          {:error, "Malformed message"}  
      end
    end

Process.spawn(receiver.(receiver))
```
The pasaboble with a function wrapped into a function is needed because of the chicken-egg problem: you need to recursively call itself from itself, which is impossible without a function capture.